### PR TITLE
[DOCU] Provide best practice guide for upgrades

### PR DESCRIPTION
**Pull request by helsner**
This is only the first step to lead new DEVs and give some hints about upgrades

### CODE REVIEW

File: docs/best_practice_guide.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# Best practice guide
+
+## What to use for
+
+You can use typo3-rector in various ways:
+- checking existing code if there are left out segments of the last upgrade
+- reformating e.g. typoscript files based on your editorconfig
```
Comment (sabbelasichon) [inline docs/best_practice_guide.md:7]: No. Rector is not reformatting all files but only changed ones

File: docs/best_practice_guide.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- partial execution of full core upgrades
+
+
+
+
+## Guide for a good upgrade
+
+
+### Starting
+Starting with an upgrade should start with installing typo3-rector and checking for the rector rules/sets of your current version, not the one you're upgrading to.
```
Comment (sabbelasichon) [inline docs/best_practice_guide.md:18]: small typo should too instead to

File: docs/best_practice_guide.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+Often there are things that were missed out in previous upgrades while rector adds rulesets for those.
+
+One thing you should definitely need for upgrades with two or more version steps is the class-alias-maps.
```
Comment (sabbelasichon) [inline docs/best_practice_guide.md:21]: This is not true anymore. I fixed it with stubs.

File: docs/best_practice_guide.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+Depending on the amount of version steps you should add the ClassAliasMap as mentioned above for e.g. v8 if you're going from v8 to v10 directly.
+
+Once again you add you're wanted/needed rulesets that should be separated by version.
+It also comes in handy to divide between TCA and TYPO3 changes AND/OR you're packages
+
+In this config example we would apply all rules of both TCA and base TYPO3 to the package_numberone:
+
+```php
+    $containerConfigurator->import(Typo3SetList::TYPO3_95);
```
Comment (sabbelasichon) [inline docs/best_practice_guide.md:60]: TYPO3_95 already includes the TCA set for that version. So every TYPO3_* contains the corresponding TCA_* set

File: docs/best_practice_guide.md (hunk 1 above)
Comment (helsner) [reply docs/best_practice_guide.md:7]: ok, will change

**Comment by sabbelasichon** [comment]
@helsner Thanks a lot. This is a very good introduction.

File: docs/best_practice_guide.md (hunk 3 above)
Comment (helsner) [reply docs/best_practice_guide.md:21]: so the whole class alias map segment is obsolete?

File: docs/best_practice_guide.md (hunk 4 above)
Comment (helsner) [reply docs/best_practice_guide.md:60]: ok. i'll rewrite that a little

File: docs/best_practice_guide.md (hunk 3 above)
Comment (sabbelasichon) [reply docs/best_practice_guide.md:21]: Yes, it is. Finally. Have a look at: https://github.com/sabbelasichon/typo3-rector/tree/main/stubs
This is taken into account with Static Reflection.

**Comment by helsner** [comment]
all done.

**Comment by sabbelasichon** [comment]
Could you add a link to this doc from the README.md

**Comment by helsner** [comment]
Done that

**Pull request merged**